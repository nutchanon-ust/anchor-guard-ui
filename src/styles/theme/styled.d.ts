import 'styled-components';
import { Theme } from './themes';

/* This is the suggested way of declaring theme types */
declare module 'styled-components' {
  export interface DefaultTheme extends Theme {}
}
