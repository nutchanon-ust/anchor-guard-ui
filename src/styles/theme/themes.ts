const lightTheme = {
  primary: 'rgba(215,113,88,1)',
  text: 'rgba(58,52,51,1)',
  textSecondary: 'rgba(58,52,51,0.7)',
  background: 'rgba(255,255,255,1)',
  backgroundVariant: 'rgba(251,249,249,1)',
  border: 'rgba(58,52,51,0.12)',
  borderLight: 'rgba(58,52,51,0.05)',
};

const darkTheme: Theme = {
  primary: 'rgba(220,120,95,1)',
  text: 'rgba(241,233,231,1)',
  textSecondary: 'rgba(241,233,231,0.6)',
  background: 'rgba(0,0,0,1)',
  backgroundVariant: 'rgba(28,26,26,1)',
  border: 'rgba(241,233,231,0.15)',
  borderLight: 'rgba(241,233,231,0.05)',
};

export type Theme = typeof lightTheme;

export const themes = {
  light: lightTheme,
  dark: darkTheme,
};
