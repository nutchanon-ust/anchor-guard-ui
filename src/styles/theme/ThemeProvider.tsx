import * as React from 'react';
import { ThemeProvider as OriginalThemeProvider } from 'styled-components';
import { useSelector } from 'react-redux';
import { useThemeSlice } from './slice';
import { selectTheme } from './slice/selectors';

export const ThemeProvider = (props: { children: React.ReactChild }) => {
  useThemeSlice();

  const theme = useSelector(selectTheme);
  return (
    <OriginalThemeProvider theme={theme}>
      {React.Children.only(props.children)}
    </OriginalThemeProvider>
  );
};
