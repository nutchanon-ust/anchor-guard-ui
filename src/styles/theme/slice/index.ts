import { PayloadAction } from '@reduxjs/toolkit';
import { createSlice } from 'utils/@reduxjs/toolkit';
import { useInjectReducer } from 'utils/redux-injectors';
import { getThemeFromStorage } from '../utils';
import { ThemeKeyType, ThemeState } from './types';

export const initialState: ThemeState = {
  selected: getThemeFromStorage() || 'system',
};

const slice = createSlice({
  name: 'theme',
  initialState,
  reducers: {
    changeTheme(state, action: PayloadAction<ThemeKeyType>) {
      state.selected = action.payload;
    },
  },
});

export const { actions: themeActions, reducer } = slice;

export const useThemeSlice = () => {
  useInjectReducer({ key: slice.name, reducer: slice.reducer });
  return { actions: slice.actions };
};
