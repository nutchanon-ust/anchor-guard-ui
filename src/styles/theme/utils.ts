import { ThemeKeyType } from './slice/types';

/* istanbul ignore next line */
export const isSystemDark = window?.matchMedia
  ? window.matchMedia('(prefers-color-scheme: dark)')?.matches
  : undefined;

export function saveTheme(theme: ThemeKeyType) {
  window.localStorage && localStorage.setItem('selectedTheme', theme);
}

/* istanbul ignore next line */
export function getThemeFromStorage(): ThemeKeyType | null {
  return window.localStorage
    ? (localStorage.getItem('selectedTheme') as ThemeKeyType) || null
    : null;
}
