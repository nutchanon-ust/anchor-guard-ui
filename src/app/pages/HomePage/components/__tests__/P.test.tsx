import * as React from 'react';
import { render } from '@testing-library/react';
import { DefaultTheme } from 'styled-components';
import { themes } from 'styles/theme/themes';
import { P } from '../P';

const renderWithTheme = (theme: DefaultTheme) => render(<P theme={theme} />);

describe('<P />', () => {
  it('should render and match the snapshot', () => {
    const p = renderWithTheme(themes.light);
    expect(p.container.firstChild).toMatchSnapshot();
  });

  it('should use theme from props', () => {
    let p = renderWithTheme(themes.light);
    expect(p.container.firstChild).toHaveStyle(
      `color: ${themes.light.textSecondary}`,
    );
    p = renderWithTheme(themes.dark);
    expect(p.container.firstChild).toHaveStyle(
      `color: ${themes.dark.textSecondary}`,
    );
  });
});
