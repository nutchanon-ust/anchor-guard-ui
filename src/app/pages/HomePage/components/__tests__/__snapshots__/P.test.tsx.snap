// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<P /> should render and match the snapshot 1`] = `
.c0 {
  font-size: 1rem;
  line-height: 1.5;
  color: rgba(58,52,51,0.7);
  margin: 0.625rem 0 1.5rem 0;
}

<p
  class="c0"
/>
`;
